core-web/apps/dotcms-ui/src/app/view/directives/dot-field-required/dot-field-required.directive.ts

```diff
@@ -0,0 +1,20 @@
+import { Directive, ElementRef, Input, OnInit, Renderer2 } from '@angular/core';
+import { FormGroupDirective } from '@angular/forms';
+
+@Directive({
+    selector: '[dotFieldRequired]'
```

Comment:
About this directive needing to be standalone, I prefer, by default add the class. If you need to check the control dynamically, use the input to do that and remove the class.

**Something like this.**

```
@Directive({
    selector: "[dotFieldRequired]",
    standalone: true
})
export class DotFieldRequiredDirective {
    @Input() for: string;

    constructor(
        private el: ElementRef,
        private renderer: Renderer2,
        private formGroupDirective: FormGroupDirective
    ) {
        renderer.addClass(this.el.nativeElement, 'p-label-input-required'); // <- always add the class
    }

    @Input()
    set checkIsRequiredControl(controlName: string) { // <- if the required is dynamic, send the control name
        if (!this.isRequiredControl(controlName)) {
            this.renderer.removeClass(this.el.nativeElement, 'p-label-input-required');
        }
    }

    private isRequiredControl(controlName: string): boolean {
        const formControl = this.formGroupDirective.control.get(controlName);

        return formControl && formControl.hasValidator(Validators.required) ? true : false;
    }

        
}
```

This allows me to do something like this.

```
<label dotFieldRequired for="endTime">
            {{ 'experiments.configure.scheduling.end.time' | dm }}
          </label>
<input formControlName='endTime' />
```
Always add the class; no need for any input to add it.

If you have a dynamic validator, use the input.

```
<label dotFieldRequired [checkIsRequiredControl]="'endTime'" for="endTime">
            {{ 'experiments.configure.scheduling.end.time' | dm }}
          </label>
<input formControlName='endTime' />
```

Reply:
Another point is, if we will use a directive to do this, we need only use the directive in all the labels with inputs required, not have, labels with the class `p-label-input-required`  and labels with directive  `dotFieldRequired`

Reply:
We used directive only for dynamic fields all the other places that we don’t have to create dynamically is not worthy,